smaps/src/main/java/com/google/smaps/Histogram.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.smaps;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Retrieves histogram information and formats it into a Json array for creating the chart
+ * and dashboard tools in histogram.js.
+ */
+@WebServlet(name = "Histogram", value = "/histogram")
+public class Histogram extends HttpServlet {
+  // Used to populate the lower bound text box and slider min.
+  static long lowerBound;
+  // Used to populate the upper bound text box and slider max.
+  static long upperBound;
```

Comment:
No initial values for lowerBound and upperBound?

Reply:
I have it so that they get set with the min and max sizes from the entire file as soon as doGet (which sets up the webpage) is called, then if a user customizes the bounds with the textboxes doPost is called and they're updated again.